core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -285,16 +285,12 @@ private[spark] object Utils extends Logging {
    */
   def createDirectory(dir: File): Boolean = {
     try {
-      // This sporadically fails - not sure why ... !dir.exists() && !dir.mkdirs()
-      // So attempting to create and then check if directory was created or not.
-      dir.mkdirs()
-      if ( !dir.exists() || !dir.isDirectory) {
-        logError(s"Failed to create directory " + dir)
-      }
-      dir.isDirectory
+      // SPARK-35907 Instead of File#mkdirs, Files#createDirectories is expected.
+      Files.createDirectories(dir.toPath)
+      true
     } catch {
       case e: Exception =>
-        logError(s"Failed to create directory " + dir, e)
+        logError(s"Failed to create directory $dir", e)
```

Comment:
This is reasonable. But, in general, I discourage to add irrelevant change like this. To have a minimal and coherent PR content is the best, @Shockang .

Reply:
> This is reasonable. But, in general, I discourage adding irrelevant changes like this. To have a minimal and coherent PR content is the best, @Shockang .

What you said is right, let me restore it, it makes sense to minimize the modification points.